### DISCUSSION TITLE
🗄️ Sobre o armazenamento de arquivos

### DISCUSSION THREAD

**Discussion by Lvght** [Lvght/e-andifes#2] [General]
**Alguns grupos vão ter que lidar com o armazenamento de arquivos no sistema**. Existem diversas formas de fazermos isso: seja usarmos BLOB ou, menos realisticamente, fazer upload para um provedor externo.

Precisamos decidir uma abordagem. Ignorar os arquivos completamente? Usar a opção simples, porém não tão boa como o BLOB? Salvar no sistema de arquivos? Provedor externo? Outra opção???

**Comment by migeyel** [comment]
Boa sorte com os arquivos. No Django do planner, eu precisei de um pacote especial só para limpar os arquivos que foram deletados do banco de dados. Isso já é difícil considerando fazer o upload de um arquivo dentro de uma transação, se a transação for abortada o arquivo precisa ser apagado, inclusive se o servidor do banco de dados for derrubado no meio do caminho. Eu não sei exatamente como funciona no servidor do planner, mas provavelmente envolve algum passo de recuperação que verifica se um arquivo "pendente" foi ou não incluído no banco de dados.
